src/tests/unit/tests/reports/components/new-tab-link-confirmation-dialog.test.tsx

```diff
@@ -1,22 +1,28 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { render } from '@testing-library/react';
 import { NewTabLink } from 'common/components/new-tab-link';
-import { shallow } from 'enzyme';
 import { isFunction } from 'lodash';
 import * as React from 'react';
 import {
     ConfirmType,
     NewTabLinkWithConfirmationDialog,
 } from 'reports/components/new-tab-link-confirmation-dialog';
-import { IMock, It, Mock, MockBehavior, Times } from 'typemoq';
+import { IMock, It, Mock, Times } from 'typemoq';
+import {
+    getMockComponentClassPropsForCall,
+    mockReactComponents,
+} from '../../../mock-helpers/mock-module-helpers';
+jest.mock('common/components/new-tab-link');
 
 describe('NewTabLinkWithConfirmationDialog', () => {
+    mockReactComponents([NewTabLink]);
     it('generates unique id for each link', () => {
-        const testSubject1 = shallow(<NewTabLinkWithConfirmationDialog />);
-        const testSubject2 = shallow(<NewTabLinkWithConfirmationDialog />);
+        const renderResult1 = render(<NewTabLinkWithConfirmationDialog />);
+        const renderResult2 = render(<NewTabLinkWithConfirmationDialog />);
```

Comment:
We won't need these variables if we access the `id` prop like the old test.
```suggestion
        render(<NewTabLinkWithConfirmationDialog />);
        render(<NewTabLinkWithConfirmationDialog />);
```

Reply:
Did the change as suggested.